src/test/java/htsjdk/samtools/cram/ref/ReferenceContextTest.java

```diff
@@ -0,0 +1,76 @@
+package htsjdk.samtools.cram.ref;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.CRAMException;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class ReferenceContextTest extends HtsjdkTest {
+
+    // test constructors and basic identity methods
+    
+    @Test
+    public void basicTest() {
+        final ReferenceContext refContext0 = new ReferenceContext(0);
+        Assert.assertTrue(refContext0.isMappedSingleRef());
+        Assert.assertFalse(refContext0.isUnmappedUnplaced());
+        Assert.assertFalse(refContext0.isMultiRef());
+        Assert.assertEquals(refContext0.getType(), ReferenceContextType.SINGLE_REFERENCE);
+        Assert.assertEquals(refContext0.getSequenceId(), 0);
+        Assert.assertEquals(refContext0.getSerializableId(), 0);
+
+        final ReferenceContext refContext1 = new ReferenceContext(1);
+        Assert.assertTrue(refContext1.isMappedSingleRef());
+        Assert.assertFalse(refContext1.isUnmappedUnplaced());
+        Assert.assertFalse(refContext1.isMultiRef());
+        Assert.assertEquals(refContext1.getType(), ReferenceContextType.SINGLE_REFERENCE);
+        Assert.assertEquals(refContext1.getSequenceId(), 1);
+        Assert.assertEquals(refContext1.getSerializableId(), 1);
+
+        final ReferenceContext unmapped = new ReferenceContext(ReferenceContext.REF_ID_UNMAPPED);
+        final ReferenceContext unmappedStatic = ReferenceContext.UNMAPPED;
+        Assert.assertEquals(unmappedStatic, unmapped);
+
+        Assert.assertFalse(unmapped.isMappedSingleRef());
+        Assert.assertTrue(unmapped.isUnmappedUnplaced());
+        Assert.assertFalse(unmapped.isMultiRef());
+        Assert.assertEquals(unmapped.getType(), ReferenceContextType.UNMAPPED_UNPLACED);
+        Assert.assertEquals(unmapped.getSerializableId(), ReferenceContext.REF_ID_UNMAPPED);
+
+        final ReferenceContext multi = new ReferenceContext(ReferenceContext.REF_ID_MULTIPLE);
+        final ReferenceContext multiStatic = ReferenceContext.MULTIPLE;
+        Assert.assertEquals(multiStatic, multi);
+
+        Assert.assertFalse(multi.isMappedSingleRef());
+        Assert.assertFalse(multi.isUnmappedUnplaced());
+        Assert.assertTrue(multi.isMultiRef());
+        Assert.assertEquals(multi.getType(), ReferenceContextType.MULTI_REFERENCE);
+        Assert.assertEquals(multi.getSerializableId(), ReferenceContext.REF_ID_MULTIPLE);
+    }
```

Comment:
I know there are a lot of existing tests written in this style, but in general we try not to gang up tests in a single method like this. These could be organized with a data provider (not sure I captured all the cases), but something like this:
```suggestion
    }    @DataProvider(name = "validRefContexts")
    private static Object[][] refContexts() {
        return new Object[][] {
                //sequenceID, expectedType, isSingle, isUnplaced, isMultiple, serializableID
                {0, ReferenceContextType.SINGLE_REFERENCE, true, false, false, 0},
                {1, ReferenceContextType.SINGLE_REFERENCE, true, false, false, 1},
                {ReferenceContext.REF_ID_MULTIPLE, ReferenceContextType.MULTI_REFERENCE, false, true, false, ReferenceContext.REF_ID_MULTIPLE},
                {ReferenceContext.REF_ID_UNMAPPED, ReferenceContextType.UNMAPPED_UNPLACED, false, false, true, ReferenceContext.REF_ID_UNMAPPED}
        };
    }

    @Test(dataProvider = "validRefContexts")
    public void referenceContextTest(
            int serializableID,
            ReferenceContextType expectedType,
            boolean expectedSingle,
            boolean expectedMulti,
            boolean expectedUnplaced,
            int expectedID) {
        final ReferenceContext refContext = new ReferenceContext(serializableID);
        Assert.assertEquals(refContext.isMappedSingleRef(), expectedSingle);
        Assert.assertEquals(refContext.isUnmappedUnplaced(), expectedUnplaced);
        Assert.assertEquals(refContext.isMultiRef(), expectedMulti);
        Assert.assertEquals(refContext.getType(), expectedType);
        Assert.assertEquals(refContext.getSerializableId(), expectedID);
    }
```